File: ocaml/acronym/acronym.ml
open Base

let acronym phrase = phrase
    |> String.split_on_chars ~on:[' '; '_'; '-']
    |> List.filter ~f:(fun s -> String.length s > 0)
    |> List.map ~f:(fun s -> Char.uppercase (String.get s 0))
    |> String.of_char_list

File: fsharp/acronym/Acronym.fs
module Acronym

open System

let abbreviate (phrase: string) =
    phrase.ToUpper().Split([| ' '; '-' |], StringSplitOptions.RemoveEmptyEntries)
    |> Array.map (Seq.filter Char.IsLetter >> Seq.head)
    |> String.Concat
